### PR TITLE
Updated kotlin version based on whitesource recommendation

### PR DESCRIPTION
**Pull request by saikaranam-amazon**
### Description
Updated kotlin version based on whitesource recommendation
- Forced `jacoco` version to `0.8.7` to work with the updated kotlin version.
 
### Issues Resolved
N/A
 
### Check List
- [x] New functionality includes testing.
  - [ ] All tests pass
- [x] New functionality has been documented.
  - [x] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by saikaranam-amazon** [comment]
security plugin load issue: https://github.com/opensearch-project/security/issues/1934

**Comment by saikaranam-amazon** [comment]
seeing jacoco test report generation failures - resulting in CI failures
```
* What went wrong:
Execution failed for task ':jacocoTestReport'.
> Error while creating report
```

**Review by ankitkala — APPROVED** [review]
LGTM

File: src/main/kotlin/org/opensearch/replication/task/CrossClusterReplicationTask.kt [hunk 1]
```diff
@@ -178,6 +179,7 @@ abstract class CrossClusterReplicationTask(id: Long, type: String, action: Strin
         client.suspending(::updatePersistentTaskState)(state)
     }
 
+    @ObsoleteCoroutinesApi
```
Comment (suranjay) [inline src/main/kotlin/org/opensearch/replication/task/CrossClusterReplicationTask.kt:182]: Why do we need this change? Can we add the reason in the description please

**Pull request merged**